contributing.md

```diff
@@ -1,9 +1,126 @@
-# Contributing to the p5.js web editor
+# Contributing to the p5.js web editor 
 
-We welcome community contributions to the p5.js web editor! You can contribute even if you are not a developer by documenting bugs, proposing new features, and updating documentation.
+Hi! Thanks for wanting to contribute to P5's web editor! You are awesome! 😊
 
-When opening an issue, be sure to clearly describe the issue, task, or feature. If it is a bug, try to include steps to reproduce or other contextual information, such as the browser or operating system. 
+This **web editor** for **p5.js**, a JavaScript library with the goal of making coding accessible to **artists**, **designers**, **educators**, and **beginners**, is used to help develop using p5.js library in web browser and thus makes for a good starting point for working with p5. 
+
+Here are links to all the sections in this document:
+
+<!-- If you change any of the headings in this document, remember to update the table of contents. -->
+
+- [Code of Conduct](#code-of-conduct)
+- [Technologies Used](#technologies-used)
+- [Development Installation](#development-installation)
+- [File Structure](#file-structure)
+- [New Design](#new-design)
+- [How Can I Contribute ?](#how-can-i-contribute?)
+  - [First Timers](#first-timers)
+  - [Want something more challenging](#want-something-more-challenging)
+  - [Feature Enhancement](#feature-enhancement)
+- [Creating a Pull request](#creating-a-pull-request)
+  - - [Tips](#tips)
+
+# Code of Conduct
+
+Please follow the guidelines mentioned at [CODE OF CONDUCT.md](https://github.com/processing/p5.js-web-editor/blob/master/CODE_OF_CONDUCT.md#p5js-code-of-conduct).
+
+# Technologies Used
+
+ > **MERN stack - MongoDB, Express, React/Redux, and Node**. 
+ 
+ - For a reference to the **file structure format** I am using, please look at the [Mern Starter](https://github.com/Hashnode/mern-starter).
+
+ - This project **does not use CSS Modules, but uses Sass**. [BEM guidelines](http://getbem.com/) are followed. 
```

Comment:
- This project **does not use CSS Modules, but uses Sass**. [BEM guidelines and naming conventions](http://getbem.com/) are followed. For repeat and common styles, write OOSCSS (Object-Oriented SCSS) with placeholders and mixins. For organizing styles, follow the [7-1 Pattern](https://sass-guidelin.es/#the-7-1-pattern) for Sass.